blaze-core/src/main/scala/org/http4s/blaze/util/ProcessWriter.scala

```diff
@@ -52,7 +53,7 @@ trait ProcessWriter {
   final private def go(p: Process[Task, ByteVector], stack: List[StackElem], cb: CBType): Unit = p match {
     case Emit(seq) if seq.isEmpty =>
       if (stack.isEmpty) writeEnd(ByteVector.empty).onComplete(completionListener(_, cb))
-      else go(Try(stack.head.apply(End).run), stack.tail, cb)
+      else Trampoline(go(Try(stack.head.apply(End).run), stack.tail, cb))
```

Comment:
OK, that makes sense. On a stylistic note, it may be easier to simply Trampoline the whole body of `go`, then all recursive calls will be trampolined automatically and developers working in this function won't have to remember to add new call sites.